src/Adapter/MSTestAdapter.PlatformServices/Execution/TestMethodInfo.cs

```diff
@@ -394,6 +392,18 @@ private async Task<TestResult> ExecuteInternalAsync(object?[]? arguments, Cancel
         {
             try
             {
+                // We invoke global test initialize methods before creating the test class instance.
+                // We consider the test class constructor as a "local" initialization.
+                // We want to invoke first the global initializations, then local ones, then test method.
+                // After that, we invoke local cleanups (including Dispose) and finally global cleanups at last.
+                foreach ((MethodInfo method, TimeoutInfo? timeoutInfo) in Parent.Parent.GlobalTestInitializations)
+                {
+                    InvokeGlobalInitializeMethod(method, ref executionContext, timeoutInfo, timeoutTokenSource);
+                }
+
+                // TODO remove dry violation with TestMethodRunner
+                _classInstance = CreateTestClassInstance(result);
```

Comment:
We need to execute this (and SetTestContext as well) on the executionContext produced by `InvokeGlobalInitializeMethod`. The only impact this will have is if the global initialize method has non-cooperative timeout.